multiplexer/abci/multiplexer.go

```diff
@@ -401,9 +401,11 @@ func openDB(rootDir string, backendType db.BackendType) (db.DB, error) {
 	return db.NewDB("application", backendType, dataDir)
 }
 
+// openTraceWriter opens a trace writer for the given file.
+// If the file is empty, it returns no writer and no error.
 func openTraceWriter(traceWriterFile string) (w io.WriteCloser, err error) {
 	if traceWriterFile == "" {
-		return w, fmt.Errorf("can not open trace writer for empty file")
+		return w, err
```

Comment:
Why not be explicit by returning `nil`?

Reply:
That's what I asked for but openTraceWriter is invoked with an empty file and then this function should be a no-op instead of erroring

Ref: https://github.com/celestiaorg/celestia-app/issues/4713#issuecomment-2843010276

Reply:
Sure that also works

Reply:
Also you can add a simple regression test to ensure that we never have that problem regress so

```go
func TestOpenTraceWriterWithEmptyFileDoesNotError(t *testing.T) {
    in := new(bytes.Buffer)
    w, err := openTraceWriter("")
    require.NoError(t, err)
    require.Equal(t, w, in)
}
```

Reply:
@rootulp kindly take a look at this test for a regression lock to prevent it from ever creeping back in.